src/hooks/useMetricData.ts

```diff
@@ -11,6 +11,15 @@ export function useMetricData(query: string, options?: (MetricQueryOptions & Use
   const [isFetchingData, setIsFetchingData] = useState(true);
   const [data, setData] = useState<any[]>([]);
   const [error, setError] = useState<string | undefined>();
+  const [refetches, setRefetches] = useState(0);
+
+  // refresh data every 60 seconds
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setRefetches(refetches + 1);
+    }, 60000);
+    return () => clearInterval(interval);
+  });
```

Comment:
**Idea**
Maybe it could be done using a single `useEffect()` hook?

```javascript
useEffect(() => {
    const getData = async () => {
      if (options?.skip) {
        return;
      }

      setIsFetchingData(true);
      const url = instance?.api?.getMetricsDS()?.url;

      if (!url) {
        return;
      }

      const { error: queryError, data: queryData } = await queryMetric(url, query, options);

      setData(queryData);
      setError(queryError);
      setIsFetchingData(false);
    };
   

    const sixtySec = 60000;
    const interval = setInterval(getData, sixtySec);

    return () => clearInterval(interval);
}, [query, instance?.api, options]);
```

Reply:
I feel more strongly about not using a seperate `useEffect()` just to force refetches than you do, @leventebalogh . ;)

There is a bug in the suggestion though (`setIsFetchingData(true)` happens too early) and overall, I'd tweak it a bit like so:
```
useEffect(() => {
  const interval = setInterval(async () => {
    const url = instance?.api?.getMetricsDS()?.url;
    if (options?.skip || !url) {
      return;
    }

    setIsFetchingData(true);

    const { error: queryError, data: queryData } = await queryMetric(url, query, options);

    setData(queryData);
    setError(queryError);
    setIsFetchingData(false);
  }, refetchInterval);

  return () => clearInterval(interval);
}, [query, instance?.api, options]);
```

... with `refetchInterval` either being declared at the top of the module or imported from a `consts` module.

(Sidenote - that `| undefined` in the `useMetricData` function signature seems unnecessary.)

Reply:
Good catch and nice tweaks! :)

Reply:
Thanks!

Reply:
Thanks y'all!